src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/Serialization/MemberRelationshipService.cs

```diff
@@ -44,7 +44,9 @@ public abstract class MemberRelationshipService
                 // and not the other as the main constructor performs argument validation.
                 if (source.Owner == null)
                 {
+#pragma warning disable CA2208 // Instantiate argument exceptions correctly
                     throw new ArgumentNullException(nameof(MemberRelationship.Owner));
```

Comment:
Similar comment here.  Seems like this should be `nameof(source.Owner)` rather than `nameof(MemberRelationship.Owner)`, and the analyzer then shouldn't warn about it because it includes an argument as part of the expression passed to nameof.

Reply:
As mentioned above we concluded to warn in `nameof(parameter.Property)` case, but maybe we should allow it for at least ArgumentNullException it seems valid for case like below: CC @terrajobst @bartonjs 
```csharp
public ChainedConfigurationProvider(ChainedConfigurationSource source)
{
    if (source == null)
    {
        throw new ArgumentNullException(nameof(source));
    }
    if (source.Configuration == null)
    {
         throw new ArgumentNullException(nameof(source.Configuration));
     }
}
```

Reply:
I was OK with the concept of reporting with the string "source.Owner".  But `nameof(source.Owner)` is just "Owner", which doesn't match a parameter name (and, in not-too-degenerate cases is ambiguous between parameters).  IIRC, the principles we took away were that paramName should be useful for tooling and that if the parameter name isn't enough, that's what the message is for.

Reply:
@terrajobst suggested ArgumentException with an appropriate message, and doc saying
![image](https://user-images.githubusercontent.com/45579687/81208345-c1046580-8f83-11ea-8101-254fb339b446.png)

so I am thinking to update this: 
`throw new ArgumentException(SR.Format(SR.InvalidNullArgument, "source.Owner"), nameof(source));`. Where the `InvalidNullArgument` message looks like  `Null is not a valid value for {0}.`